### ISSUE TITLE
[Bug]: CCE in BLangNodeBuilder for QualifiedNameReferenceNode

### ISSUE THREAD

**Issue by IMS94** [ballerina-platform/ballerina-lang#38886]
### Description

Consider the following incomplete code
```ballerina
final websocket:Client servicesClient = check new (<cursor>wss://" + k8sHost + "/services",
auth = {
    token: token
},
secureSocket = {
    cert: caCertPath
});
```
At cursor, I haven't typed the `"` yet. Following error is shown in LS logs and when `bal build` is run.
```java
Compiling source
        tmp.bal
ballerina: Oh no, something really went wrong. Bad. Sad.

We appreciate it if you can report the code that broke Ballerina in
https://github.com/ballerina-platform/ballerina-lang/issues with the
log you get below and your sample code.

We thank you for helping make us better.

[2022-11-30 14:06:58,650] SEVERE {b7a.log.crash} - class io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode cannot be cast to class io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode (io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode and io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode are in unnamed module of loader 'app') 
java.lang.ClassCastException: class io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode cannot be cast to class io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode (io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode and io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode are in unnamed module of loader 'app')
        at io.ballerina.compiler.syntax.tree.NamedArgumentNode.argumentName(NamedArgumentNode.java:36)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:3066)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:476)
        at io.ballerina.compiler.syntax.tree.NamedArgumentNode.apply(NamedArgumentNode.java:54)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createActionOrExpression(BLangNodeBuilder.java:5243)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createExpression(BLangNodeBuilder.java:5216)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.populateArgsInvocation(BLangNodeBuilder.java:2215)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createInvocation(BLangNodeBuilder.java:2202)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:2165)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:476)
        at io.ballerina.compiler.syntax.tree.ImplicitNewExpressionNode.apply(ImplicitNewExpressionNode.java:51)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createActionOrExpression(BLangNodeBuilder.java:5243)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createExpression(BLangNodeBuilder.java:5216)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:1921)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:476)
        at io.ballerina.compiler.syntax.tree.CheckExpressionNode.apply(CheckExpressionNode.java:50)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createActionOrExpression(BLangNodeBuilder.java:5243)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createExpression(BLangNodeBuilder.java:5216)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.initializeBLangVariable(BLangNodeBuilder.java:2906)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:632)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:476)
        at io.ballerina.compiler.syntax.tree.ModuleVariableDeclarationNode.apply(ModuleVariableDeclarationNode.java:71)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:610)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:476)
        at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
        at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.accept(BLangNodeBuilder.java:509)
        at io.ballerina.projects.DocumentContext.compilationUnit(DocumentContext.java:111)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:418)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:380)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:203)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:187)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:99)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:94)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:233)
        at io.ballerina.projects.Package.getCompilation(Package.java:148)
        at io.ballerina.projects.Package.runCodeGenAndModifyPlugins(Package.java:256)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:126)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:256)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
 
```

### Affected Version(s)

2201.2.3

### Related area

-> Compilation

**Comment by lochana-chathura** [comment]
Seems this has the same root cause as #39581

**Comment by lochana-chathura** [comment]
We merged the fix for #39581 to master just now. @DamithaSenevirathne Can you verify this sample in the master now?

**Comment by DamithaSenevirathne** [comment]
This issue is fixed by https://github.com/ballerina-platform/ballerina-lang/issues/39581
